src/libSchnorr/src/Schnorr_PubKey.cpp

```diff
@@ -112,8 +112,9 @@ bool PubKey::Serialize(bytes& dst, unsigned int offset) const {
 }
 
 bool PubKey::Deserialize(const bytes& src, unsigned int offset) {
+
   shared_ptr<EC_POINT> result =
-      ECPOINTSerialize::GetNumber(src, offset, PUB_KEY_SIZE);
+      ECPOINTSerialize::GetNumber(src, offset, src.size());
```

Comment:
when the size of src was not PUB_KEY_SIZE this would fail. However if you allow compressed and uncompressed signatures your pub key size could vary.